pxr/usd/sdf/textFileFormat.ll

```diff
@@ -134,24 +164,42 @@ PXR_NAMESPACE_USING_DIRECTIVE
 "variants"            { (*yylval_param) = std::string(yytext, yyleng); return TOK_VARIANTS; }
 "varying"             { (*yylval_param) = std::string(yytext, yyleng); return TOK_VARYING; }
 
-    /* unquoted C/Python identifier */
-[[:alpha:]_][[:alnum:]_]* {
-        (*yylval_param) = std::string(yytext, yyleng);
-        return TOK_IDENTIFIER;
-    }
-
-    /* unquoted C++ namespaced identifier -- see bug 10775 */
-[[:alpha:]_][[:alnum:]_]*(::[[:alpha:]_][[:alnum:]_]*)+ {
-        (*yylval_param) = std::string(yytext, yyleng);
-        return TOK_CXX_NAMESPACED_IDENTIFIER;
-    }
-
-    /* unquoted namespaced identifier.  matches any number of colon
-     * delimited C/Python identifiers */
-[[:alpha:]_][[:alnum:]_]*(:[[:alpha:]_][[:alnum:]_]*)+ {
-        (*yylval_param) = std::string(yytext, yyleng);
-        return TOK_NAMESPACED_IDENTIFIER;
-    }
+ /* In a Unicode enabled scheme, 'identifiers' are generally
+  * categorized as something that begins with something in the
+  * XID_Start category followed by zero or more things in the
+  * XID_Continue category.  Since the number of characters in
+  * these classes are large, we can't explicitly validate them
+  * here easily, so the lex rule is pretty permissive with some
+  * further validation done in code prior to calling what was
+  * read an 'identifier'.  Note this rule will also match
+  * standard ASCII strings because the UTF-8 encoded byte 
+  * representation is the same for these characters.
+  * However, unlike the path lexer, we can guarantee that 
+  * prim names aren't something special to be called out here
+  * so we can be a little more specific about the kinds of strings
+  * we match, particularly to not collide with the pure digit match rule
+  * below
+  */
+{UTF8NODIGU}{UTF8U}* {
```

Comment:
It seems like there ought to be code added to the text file parser to actually validate that identifiers are XID_Start/XID_Continue -- is that coming in a future change?

Reply:
Alex, are you thinking that Ed should call `SdfPath::IsValidIdentifier` and `SdfPath::IsValidNamespacedIdentifier` here?

Reply:
The original changes to the path lexer did indeed have to do an extra validation check before reducing as a `TOK_IDENTIFIER`.  The text file format parser did not - I can go back and verify why.

Reply:
Added additional validation to the three lexer rules